pkg/provisioner/ironic/ironic.go

```diff
@@ -310,7 +322,7 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 
 			// If the node has a name, this means we didn't find it above.
 			if ironicNode.Name != "" {
-				return nil, errors.New(fmt.Sprint("node found by MAC but has a name: ", ironicNode.Name))
+				return nil, NewMacAddressConflict(p.host.Spec.BootMACAddress)
```

Comment:
Can we provide enough information to tell which other host has the same MAC?

Reply:
We can add the name of the node we found, good idea